simphony/material_relations/material_relation.py

```diff
@@ -0,0 +1,118 @@
+from simphony.material_relations.abc_material_relation import (
+    ABCMaterialRelation)
+import simphony.core.data_container as dc
+
+
+class MaterialRelation(ABCMaterialRelation):
+    """ MaterialRelation provides a general interface for describing the
+    (physics/chemistry) relations between different materials
+
+    Attributes
+    ----------
+    name : str
+        name of the material-relation
+
+    description: str
+        user-defined description of the material-relation
+
+    parameters: DataContainer
+        the required parameters
+
+    supported_parameters: list of CUBA
+        CUBA values required/allowed for the parameters
+
+    materials: list of uids
+        materials where this relation applies
+
+    num_materials: list of int
+        list with all possible configurations of avaliable
+        number of materials in the relation
+
+    kind: CUBA
+        Describes the kind of the MaterialRelation
+
+    Raises
+    ------
+    ValueError :
+        If the number of materials does not match with any value of
+        num_materials
+
+    """
+
+    def __init__(self, name, description, parameters, supported_parameters,
+                 materials, num_materials, kind):
+
+        self._num_materials = num_materials
+
+        if(len(materials) not in self._num_materials):
+            error_str = "Incorrect number of materials, expected: {}"
+            raise ValueError(
+                error_str.format(
+                    materials.size(),
+                    self._num_materials
+                )
+            )
+
+        self._name = name
+        self._description = description
+        self._parameters = parameters
+        self._supported_parameters = supported_parameters
+        self._materials = materials
+        self._kind = kind
+
+    # Should we allow users to modify the id?
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
```

Comment:
A CUDS (to avoid confusion: see below) will have many different material relations. The user needs a convinient way to check which relations are defined, and to be able to delete, update them etc. A name, as a tag that will allow the user to identify the material relation is therefore a must and is very useful. internally we can let every thing work with uuids if it is really going to make the code easier. But uuids are not user friendly (despite being the users best friend!)

CUDS: this is the new (original) CUDS object, that will be independent of an engine, and store all information relevant for one computational model..)